eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Models/ResourceMetadata.cs

```diff
@@ -19,6 +19,71 @@ internal record ResourceMetadata(
         string? ParentResourceId,
         IReadOnlyList<string> ChildResourceIds)
     {
+        // Deserialize from the new unified @armProviderSchema decorator
+        internal static ResourceMetadata DeserializeResourceMetadataFromArmProviderSchema(JsonElement element, InputModelType inputModel, IReadOnlyList<string> childResourceIds)
```

Comment:
I think our new method should replace everywhere that calls the old method below.
Please remove the below method and let the caller of it call this method instead.
Then finally, this new method should use the name of that old method's